ftw/solr/query.py

```diff
@@ -75,3 +80,78 @@ def make_query(phrase):
     if settings.local_query_parameters:
         query = settings.local_query_parameters + query
     return query
+
+
+def make_path_filter(path, depth=0, include_self=False):
+    filters = []
+    if depth == 0:
+        filters.append(u'path:{}'.format(ensure_text(escape(path))))
+    else:
+        filters.append(u'path_parent:{}'.format(ensure_text(escape(path))))
+    if depth > 0:
+        current_depth = len(path.split('/')) - 1
+        min_depth = current_depth if include_self else current_depth + 1
+        max_depth = current_depth + depth
+        if max_depth == min_depth:
+            filters.append(u'path_depth:{}'.format(max_depth))
+        else:
+            filters.append(u'path_depth:[{} TO {}]'.format(
+                min_depth, max_depth))
+    return filters
+
+
+def make_filters(**kwargs):
+    manager = getUtility(ISolrConnectionManager)
+    filters = []
+    for key, value in kwargs.items():
+        if key not in manager.schema.fields:
+            logger.warning(
```

Comment:
we've had that discussion before, if this is used in gever we'd need a way to log to harvest directly. as an alternative you could also raise an exception here if **kwargs are not valid.

Reply:
That's still possible, but you have to do it in gever. Will require an additional check though.